Sources/MongoSwift/MongoCollection+Write.swift

```diff
@@ -88,23 +72,12 @@ extension MongoCollection {
     public func replaceOne(filter: Document,
                            replacement: CollectionType,
                            options: ReplaceOptions? = nil) throws -> UpdateResult? {
-        let replacementDoc = try self.encoder.encode(replacement)
-        let opts = try self.encoder.encode(options)
-        let reply = Document()
-        var error = bson_error_t()
-        guard mongoc_collection_replace_one(
-            self._collection, filter.data, replacementDoc.data, opts?.data, reply.data, &error) else {
-            throw getErrorFromReply(bsonError: error, from: reply)
-        }
-
-        guard isAcknowledged(options?.writeConcern) else {
-            return nil
-        }
-
-        return try self.decoder.internalDecode(
-                UpdateResult.self,
-                from: reply,
-                withError: "Couldn't understand response from the server.")
+        let operation = UpdateOperation(collection: self,
+                                        filter: filter,
+                                        update: try self.encoder.encode(replacement),
```

Comment:
should we be more explicit with throwable methods by calling the method outside of the operation initializer?
```
let updateDoc = try self.encoder.encode(replacement)
let operation = UpdateOperation(...
```